axelor-account/src/main/java/com/axelor/apps/account/service/invoice/generator/tax/TaxInvoiceLine.java

```diff
@@ -274,22 +301,30 @@ protected List<InvoiceLineTax> finalizeInvoiceLineTaxes(
   protected void computeAndAddInvoiceLineTax(
       InvoiceLineTax invoiceLineTax,
       List<InvoiceLineTax> updatedInvoiceLineTaxList,
-      List<InvoiceLineTax> invoiceLineTaxList) {
-    BigDecimal taxValue =
-        invoiceLineTax
-            .getTaxLine()
-            .getValue()
-            .divide(
-                BigDecimal.valueOf(100), AppBaseService.COMPUTATION_SCALING, RoundingMode.HALF_UP);
-
+      List<InvoiceLineTax> invoiceLineTaxList,
+      BigDecimal sumOfAllDeductibleRateValue,
+      BigDecimal sumOfAllNonDeductibleRateValue) {
+
+    Boolean isNonDeductibleTax = invoiceLineTax.getTaxLine().getTax().getIsNonDeductibleTax();
+    BigDecimal originalTaxRateValue = invoiceLineTax.getTaxLine().getValue();
+    BigDecimal adjustedTaxValue =
+        Beans.get(TaxService.class)
```

Comment:
This injection should be done in the constructor to avoid doing it on each iteration of the loop